language/tools/test_generation/src/abstract_state.rs

```diff
@@ -123,15 +135,18 @@ impl AbstractState {
     pub fn from_locals(
         module: CompiledModuleMut,
         locals: HashMap<usize, (AbstractValue, BorrowState)>,
+        acquires_global_resources: Vec<StructDefinitionIndex>,
     ) -> AbstractState {
         AbstractState {
             stack: Vec::new(),
-            locals,
+            locals: locals.clone(),
```

Comment:
Rather than clone locals here (not cheap), precompute and cache locals.len() (cheap).